nselib/httpcookies.lua

```diff
@@ -0,0 +1,476 @@
+        end
+          elseif(cookie_attribute == 'value') then
+          if(type(cookie_attribute_value) ~= 'string') then
+            stdnse.debug1("Cookie value is not a string")
+            return false
+          end
+          elseif(cookie_attribute == 'path') then
+            if(type(cookie_attribute_value) ~= 'string') then
+              stdnse.debug1("Cookie path is not a string")  
+            return false
+          end
+        elseif(cookie_attribute == 'expires') then
+          if(type(cookie_attribute_value) ~= 'string') then
+            stdnse.debug1("Cookie expires attribute is not a string") 
+            return false
+          end
+        elseif(cookie_attribute == 'max-age') then
+          if(type(cookie_attribute_value) ~= 'string') then
+            stdnse.debug1("Cookie max-age attribute is not a string") 
+            return false
+          end
+        elseif(cookie_attribute == 'domain') then
+          if(type(cookie_attribute_value) ~= 'string') then
+            stdnse.debug1("Cookie domain attribute is not a string")  
+            return false
+          end
+        elseif(cookie_attribute == 'secure') then 
+          if(type(cookie_attribute_value) ~= 'boolean') then
+            stdnse.debug1("Cookie secure attribute is not a boolean")
+            return false
+          end
+        elseif(cookie_attribute == 'httponly') then
+          if(type(cookie_attribute_value) ~= 'boolean') then
+            stdnse.debug1("Cookie httponly attribute is not a boolean")
+            return false
+          end
+        else 
+          stdnse.debug1("Cookie attribute is not recognised")
+          return false
+        end
+      end
+    end 
+    return true
+  elseif (type(cookies) == 'string') then 
+      --We can parse it using the http parse cookie function
+      cookies, status = http.parse_set_cookie(cookies)
+      --Does it make sense to have multiple cookies parsed when we are using a string argument(?)
+      if cookies == nil then
+        return false
+      end
+      return true, cookies
+    else
+      return false--A cookie can only be string or table
+    end
+  end,
+
+  --- This function takes the <code>host</code>, <code>port</code> and <code>path</code>
+  -- and for each cookie table in the cookiejar, it checks for the attributes correctly
+  -- and then adds it to the cookie_table.
+  -- @param host Host table
+  -- @param port Port table
+  -- @param path Path for which the get function is called.
+  -- @return cookies The complete cookie table which considers all the attributes and
+  -- sends a cookiejar taking only the eligible cookies into consideration
+  check_cookie_attributes = function(self, host, port, path)
+    local cookie_table = {}
+    --Changing type of port to table cause shortport library expects a port table
+    if type(port) == 'number' then
+      port = {port}
+    end
+    local host_name
+    if not string.match(host, "^%d") then
```

Comment:
I think this is incorrect as well because host is a table, not a string. This might be working on your side because of how you are testing. I suggest you write the script we discussed implementing the library to check that everything works.

Reply:
The new script was implemented after this comment. So, I guess this is correct. Besides I checked that host is a string. 
https://nmap.org/nsedoc/lib/stdnse.html .Since get_hostname() returns a string and http.host used the output of stdnse.get_hostname(). Please check!